pkgs/applications/graphics/imv/default.nix

```diff
@@ -1,61 +1,116 @@
-{ asciidoc
+{ stdenv
+, lib
+, fetchFromGitHub
+, asciidoc
 , cmocka
 , docbook_xsl
-, fetchFromGitHub
+, libxslt
 , fontconfig
-, freeimage
+, meson
+, ninja
+, pkgconfig
 , icu
+, pango
+, inih
+, withWindowSystem ? "all"
+, xorg
+, libxkbcommon
 , libGLU
-, libheif
+, wayland
+, withFreeimage ? true
```

Comment:
Uhm, these are a lot of flags. It would be easier if you could do something like
```
imv.override { formats = [ "jpeg" "png" "svg" ] };
```
It could be implemented with 
```
-Dlibjpeg=${boolToEnable (elem "jpeg" formats)}
```

Reply:
Might make for a bit of ugly asserts, but I think we can rework this into an `backends` parameter.

Reply:
Done in 5d57a6b4dde41022650f0a9a7ff6f87799090150. Error messages for wrong `withWindowSystem` or `withBackends` values should be a lot more helpful now thanks to `lib.assertOneOf`.